Sources/NIOPerformanceTester/main.swift

```diff
@@ -833,4 +833,71 @@ try measureAndPrint(desc: "lock_4_threads_10M_ops",
 try measureAndPrint(desc: "lock_8_threads_10M_ops",
                     benchmark: LockBenchmark(numberOfThreads: 8, lockOperationsPerThread: 1_250_000))
 
+// We are preheating the ETL to avoid growing the `ScheduledTask` `PriorityQueue`
+// during the actual test
+let loop = group.next()
+try! loop.submit {
+    var counter: Int = 0
+    for _ in 0..<100000 {
+        loop.scheduleTask(in: .nanoseconds(0)) {
+            counter += 1
+        }
+    }
+}.wait()
+
+measureAndPrint(desc: "schedule_10000_tasks") {
+    let counter = try! loop.submit { () -> Int in
+        var counter: Int = 0
+        for _ in 0..<10000 {
+            loop.scheduleTask(in: .hours(1)) {
+                counter += 1
+            }
+        }
+
+        return counter
+    }.wait()
+
+    return counter
+}
+
+measureAndPrint(desc: "schedule_and_run_10000_tasks") {
+    let dg = DispatchGroup()
+
+    var counter = 0
+
+    try! loop.submit {
+        for _ in 0..<10000 {
+            dg.enter()
+
+            loop.scheduleTask(in: .nanoseconds(0)) {
+                counter += 1
+                dg.leave()
+
+            }
+        }
+    }.wait()
+    dg.wait()
+
+    return counter
+}
+
+measureAndPrint(desc: "execute_10000") {
+    let dg = DispatchGroup()
+
+    var counter = 0
 
+    try! loop.submit {
+        for _ in 0..<10000 {
+            dg.enter()
+
+            loop.execute {
+                counter += 1
+                dg.leave()
+
+            }
+        }
+    }.wait()
+    dg.wait()
+
+    return counter
+}
```

Comment:
Can I ask that you use the `Benchmark` class instead of throwing these things into top-level tests? It's a bit of an annoyance, but this file is rapidly growing unmanageable.

Reply:
Agreed. I was actually wondering what the proper place for this would be.

Reply:
You can just create new files alongside this one.